src/PromiseInterface.php

```diff
@@ -2,6 +2,7 @@
 
 namespace React\Promise;
 
+/** @template T */
```

Comment:
Should we have a `TFulfilled` and a `TRejected` as discussed in https://github.com/phpstan/phpstan/discussions/7371?

Reply:
Playing with that locally at the moment to get the types checking in CI as you mentioned above.

Reply:
> After the discussion below, perhaps we should indeed disregard my comment and focus solely on `TFulfilled` instead? ([#227 (comment)](https://github.com/reactphp/promise/pull/227#issuecomment-1170143308))

Done!

> Sorry for the confusion!

No worries, we needed that discussion 😃